src/config/ts-node.ts

```diff
@@ -15,17 +15,43 @@ const debug = Debug('ts-node')
 export const TS_CONFIGS: Record<string, TSConfig> = {}
 const REGISTERED = new Set<string>()
 
+function importTypescript(root: string) {
+  let typescript: typeof import('typescript') | undefined
+  try {
+    typescript = require('typescript')
+  } catch {
+    try {
+      typescript = require(require.resolve('typescript', {paths: [root, __dirname]}))
+    } catch {
+      debug(`Could not find typescript dependency. Skipping ts-node registration for ${root}.`)
+      memoizedWarn(
+        'Could not find typescript. Please ensure that typescript is a devDependency. Falling back to compiled source.',
+      )
+      return
+    }
+  }
+
+  return typescript
+}
+
 async function loadTSConfig(root: string): Promise<TSConfig | undefined> {
   try {
     if (TS_CONFIGS[root]) return TS_CONFIGS[root]
+
+    const typescript = importTypescript(root)
+    if (!typescript) return
+
     const tsconfigPath = join(root, 'tsconfig.json')
-    const tsconfig = await readJson<TSConfig>(tsconfigPath)
+    const raw = await safeReadFile(tsconfigPath)
+    if (!raw) return
 
-    if (!tsconfig || Object.keys(tsconfig.compilerOptions).length === 0) return
+    const {config} = typescript.parseConfigFileTextToJson(tsconfigPath, raw)
 
-    TS_CONFIGS[root] = tsconfig
+    if (!config || Object.keys(config.compilerOptions).length === 0) return
 
-    if (tsconfig.extends) {
+    TS_CONFIGS[root] = config
+
+    if (config.extends) {
       const {parse} = await import('tsconfck')
```

Comment:
TIL about `tsconfck`

q: could we `tsconfck` to parse the whole tsconfig instead of just the `extends` block?